cscs-checks/apps/greasy/greasy_check.py

```diff
@@ -0,0 +1,218 @@
+
+        for output_file in output_files:
+            result = sn.findall(r'Hello, World from thread \s*(\d+) out '
+                                r'of \s*(\d+) from process \s*(\d+) out of '
+                                r'\s*(\d+)', output_file)
+
+            failure_msg = ('Found %s Hello, World... pattern(s), but expected '
+                           '%s pattern(s) inside the output file %s' % (
+                                              sn.count(result),
+                                              num_tasks * num_cpus_per_task,
+                                              output_file))
+            sn.evaluate(sn.assert_eq(sn.count(result),
+                                     num_tasks * num_cpus_per_task,
+                                     msg=failure_msg))
+
+            sn.evaluate(sn.assert_true(sn.all(
+                sn.chain(
+                    sn.map(lambda x: sn.assert_lt(tid(x), num_threads(x)),
+                            result),
+                    sn.map(lambda x: sn.assert_lt(rank(x), num_ranks(x)),
+                            result),
+                    sn.map(
+                        lambda x: sn.assert_lt(tid(x), num_cpus_per_task),
+                        result),
+                    sn.map(
+                        lambda x: sn.assert_eq(num_threads(x),
+                                                num_cpus_per_task),
+                        result),
+                    sn.map(lambda x: sn.assert_lt(rank(x), num_tasks),
+                            result),
+                    sn.map(lambda x: sn.assert_eq(num_ranks(x), num_tasks),
+                            result),
+                )
+            )))
+
+        sn.evaluate(sn.assert_eq(sn.count(sn.findall(r'Finished greasing',
+                                         self.greasy_logfile)), 1))
+
+        result = sn.findall(r'INFO: Summary of (\d+) tasks: '
+                            r'(\d+) OK, '
+                            r'(\d+) FAILED, '
+                            r'(\d+) CANCELLED, '
+                            r'(\d+) INVALID\.', output_file)
+        sn.evaluate(sn.assert_true(sn.all(
+                    sn.chain(
+                        sn.map(lambda x: sn.assert_eq(int(x.group(1)),
+                                                      self.num_greasy_tasks),
+                               result),
+                        sn.map(lambda x: sn.assert_eq(int(x.group(2)),
+                                                      self.num_greasy_tasks),
+                               result),
+                        sn.map(lambda x: sn.assert_eq(int(x.group(3)), 0),
+                               result),
+                        sn.map(lambda x: sn.assert_eq(int(x.group(4)), 0),
+                               result),
+                        sn.map(lambda x: sn.assert_eq(int(x.group(5)), 0),
+                               result),
+                    )
+                )))
```

Comment:
I don't understand very well this test. Do you have multiple summary lines in the output? And why you need to look separately for `0 FAILED` in the assert function instead of requiring that explicitly in the regex that retrieves the `result`?